test/units/testsuite-50.sh

```diff
@@ -363,6 +363,11 @@ ExecStart=/bin/sh -c ' \\
 EOF
 systemctl start testservice-50d.service
 
+# Mount twice to exercise mount-beneath (on kernel 6.5+, on older kernels it will just overmount)
+mkdir -p /tmp/wrong/foo
+mksquashfs /tmp/wrong/foo /tmp/wrong.raw
+systemctl mount-image --mkdir testservice-50d.service /tmp/wrong.raw /tmp/img
+test "$(systemctl show -P SubState testservice-50d.service)" = "running"
```

Comment:
So that seems a bit sparse to me.
I think for code as sensitive as this it would be nice to verify that on kernels that do support MOVE_MOUNT_BENEATH the prior mount is actually gone.